### PR TITLE
Feature/129450 validation

### PR DESCRIPTION
**Pull request by Camwyn**
Validation.
AR Page submission.
Non-modal ticket submission.

🎫 https://central.tri.be/issues/134052
🎫 https://central.tri.be/issues/129435 (sorta)

### CODE REVIEW

File: src/views/blocks/tickets/quantity-number.php [hunk 1]
```diff
@@ -24,6 +24,7 @@
 
 /** @var Tribe__Tickets__Tickets_Handler $handler */
 $handler      = tribe( 'tickets.handler' );
+bdump($ticket->ID);
```
Comment (sc0ttkclark) [inline src/views/blocks/tickets/quantity-number.php]:
```suggestion
```

File: src/views/blocks/tickets/submit-button.php [hunk 2]
```diff
@@ -15,10 +15,15 @@
  * @version TBD
  *
  */
+$button_name = apply_filters( 'tribe_tickets_submit', 'cart-button' );
```
Comment (sc0ttkclark) [inline src/views/blocks/tickets/submit-button.php]:
```suggestion
$button_name = apply_filters( 'tribe_tickets_ticket_block_submit', 'cart-button' );
```

File: src/resources/js/tickets-registration-page.js [hunk 3]
```diff
@@ -472,6 +469,61 @@ console.log(value);
...
+	/**
+	 * Get the Currency Formatting for a Provider.
+	 *
+	 * @since TBD
+	 *
+	 * @returns {*}
+	 */
+	obj.getCurrencyFormatting = function () {
+		var currency = JSON.parse( TribeCurrency.formatting );
+		var format   = currency[ obj.commerceSelector[obj.providerId] ];
```
Comment (sc0ttkclark) [inline src/resources/js/tickets-registration-page.js]:
```suggestion
		var format   = currency[ obj.commerceSelector[ obj.providerId ] ];
```

File: src/resources/js/tickets-registration-page.js [hunk 4]
```diff
@@ -526,6 +578,68 @@ console.log(value);
...
+			var params = {
+				provider: obj.providerId,
+				tickets : {},
+				meta    : obj.getMetaForSave(),
+				post_id : obj.postId,
+			};
+
+			$.ajax({
+				type: 'POST',
+				url: '/wp-json/tribe/tickets/v1/cart',
```
Comment (sc0ttkclark) [inline src/resources/js/tickets-registration-page.js]: Be aware that REST API prefixes can be customized, we should leverage something to account for this.

File: src/resources/js/tickets-registration-page.js [hunk 5]
```diff
@@ -526,6 +578,68 @@ console.log(value);
...
+					var url = response.checkout_url;
+					if ( undefined === url || ! url ) {
+						return false;
+					} else {
+					}
+
+					window.location.href = url;
+				},
+				fail: function( response ) {
+					// @TODO: add messaging on error?
```
Comment (sc0ttkclark) [inline src/resources/js/tickets-registration-page.js]: Handle error yet?

Comment (Camwyn) [reply src/resources/js/tickets-registration-page.js]: Nope - I need some direction there I think. Do we want to display it to the user?

Comment (Camwyn) [reply src/resources/js/tickets-registration-page.js]: Will be covered by https://central.tri.be/issues/134944

File: src/resources/js/tickets-block.js [hunk 6]
```diff
@@ -1312,7 +1328,56 @@ tribe.tickets.block = {
...
+	obj.document.on(
+		'click',
+		obj.selector.submit,
+		function( e ) {
+			e.preventDefault();
+			var $button    = $( this );
+
+			// save meta and cart
+			var params = {
+				provider: obj.commerceSelector[obj.tribe_ticket_provider],
```
Comment (sc0ttkclark) [inline src/resources/js/tickets-block.js]:
```suggestion
				provider: obj.commerceSelector[ obj.tribe_ticket_provider ],
```

**Pull request merged**